### ISSUE TITLE
Repeadly calling `->reload` leads to "request too large" if there are cookies

### ISSUE THREAD

**Issue by dakkar** [libwww-perl/WWW-Mechanize#52]
Relevant facs:
* `LWP::UserAgent` sets up a `request_prepare` handler that calls `$self->cookie_jar->add_cookie_header($req)`
* `HTTP::Cookies::add_cookie_header` always preserves existing cookies in a request object
* `WWW::Mechanize::reload` calls `$self->request($self->{req})`

As a consequence, for every reload, the length of the `Cookie` header increases, until the server refuses to fulfil the request.

It can be argued that `HTTP::Cookies` should merge existing cookies, not just add. I'll be sending a bug report for that module, as well.

In the meantime, a simple workaround in `WWW::Mechanize` is:
```perl
sub reload {
    my $self = shift;

    return unless my $req = $self->{req};
    $req->remove_header('Cookie');

    return $self->_update_page( $req, $self->_make_request( $req, @_ ) );
}
```

Hope this helps.

**Comment by dakkar** [comment]
Turns out there's already a bug report for `HTTP::Cookies`: https://rt.cpan.org/Public/Bug/Display.html?id=75897
It's been there for 4 years.

**Comment by oalders** [comment]
Yikes! We can at least quickly get this fixed in `WWW::Mechanize`.  Are you able to put a pull request with a relevant test together?

**Comment by oalders** [comment]
Closed via #53
